packages/keystone/bin/loading.html

```diff
@@ -0,0 +1,162 @@
+      }
+      .lds-ellipsis div:nth-child(1) {
+        left: 6px;
+        animation: lds-ellipsis1 0.6s infinite;
+      }
+      .lds-ellipsis div:nth-child(2) {
+        left: 6px;
+        animation: lds-ellipsis2 0.6s infinite;
+      }
+      .lds-ellipsis div:nth-child(3) {
+        left: 26px;
+        animation: lds-ellipsis2 0.6s infinite;
+      }
+      .lds-ellipsis div:nth-child(4) {
+        left: 45px;
+        animation: lds-ellipsis3 0.6s infinite;
+      }
+      @keyframes lds-ellipsis1 {
+        0% {
+          transform: scale(0);
+        }
+        100% {
+          transform: scale(1);
+        }
+      }
+      @keyframes lds-ellipsis3 {
+        0% {
+          transform: scale(1);
+        }
+        100% {
+          transform: scale(0);
+        }
+      }
+      @keyframes lds-ellipsis2 {
+        0% {
+          transform: translate(0, 0);
+        }
+        100% {
+          transform: translate(19px, 0);
+        }
+      }
+    </style>
+  </head>
+  <body>
+    <div class="container">
+      <div class="lds-ellipsis">
+        <div></div>
+        <div></div>
+        <div></div>
+        <div></div>
+      </div>
+      <h1>Keystone is <span id="status">loading...</span></h1>
+      (This page will reload when the server is ready)
+    </div>
+    <script>
+      const statusEl = document.querySelector('#status');
+      let ready = false;
+
+      function onReady() {
+        statusEl.innerHTML = 'ready 🚀';
+        ready = true;
+        location.reload(true);
+      }
+
+      function checkAndSetStatus() {
+        let cancelled = false;
+        fetch('/', { headers: { Accept: 'application/json' } })
+          .then(result => result.json())
+          .catch(error => {
+            // We can get back an error "Cannot parse JSON" when a HTML
+            // response is returned, so we assume the server is ready
+            onReady();
+          })
+          .then(({ loading, status } = {}) => {
+            if (!loading) {
+              onReady();
+              return;
+            }
+            if (cancelled) {
+              return;
+            }
+            switch (status) {
+              case 'init-keystone': {
+                statusEl.innerHTML = 'initialising...';
+                break;
+              }
+              case 'db-connect': {
+                statusEl.innerHTML = 'connecting to the DB...';
+                break;
+              }
+              case 'start-server':
+              default: {
+                statusEl.innerHTML = 'loading...';
+                break;
+              }
+            }
+          });
+        return () => {
+          cancelled = true;
+        };
+      }
+
+      let cancelLastCall = checkAndSetStatus();
+      const interval = setInterval(() => {
+        cancelLastCall();
+        if (ready) {
+          clearInterval(interval);
+          return;
+        }
+        cancelLastCall = checkAndSetStatus();
+      }, 500);
```

Comment:
imo we could make this faster; like every 120ms. the load is trivial and the current setup could lead to up to 1/2s wait _longer_ than necessary before bouncing into the initialised app, making startup feel slower than it is.